packages/core/src/baseclient.ts

```diff
@@ -689,30 +688,18 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
         return evt;
       }
 
-      const propagationContext = {
-        ...isolationScope.getPropagationContext(),
-        ...(currentScope ? currentScope.getPropagationContext() : undefined),
+      evt.contexts = {
+        trace: getTraceContextFromScopes(currentScope, isolationScope),
+        ...evt.contexts,
       };
 
-      const trace = evt.contexts && evt.contexts.trace;
-      if (!trace && propagationContext) {
```

Comment:
These checks were not really necessary, I believe - first, `propagationContext` cannot be empty here. And second, we do not need to check for `trace` as this will be overwritten anyhow by `...evt.contexts` if it already exists.